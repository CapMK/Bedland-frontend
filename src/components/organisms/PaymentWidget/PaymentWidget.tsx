import routes from 'App/routing/routes';
import CustomChart from 'components/molecules/CustomChart/CustomChart';
import Widget from 'components/templates/Widget/Widget';
import { useAppSelector } from 'store/hooks';
import { useNavigate } from 'react-router-dom';
import { PaymentChartInputDataType } from '../../molecules/CustomChart/CustomChart.types';

const dataForPaymentChartManager: PaymentChartInputDataType = {
  'Paid on time': 75,
  'Awaiting payments': 21,
  'Payment overdue': 4,
};

const dataForPaymentChartResident = {
  'Awaiting payments': 25,
  'Paid on time': 75,
};

const PaymentWidget = () => {
  const userRole = useAppSelector((state) => state.user.role);
  const navigate = useNavigate();

  return (
    <div>
      <Widget
        variant='payments'
        title='Payments'
        mode='light'
        onClick={() => navigate(routes.payments)}
        isArrowButton
        data-testid='paymentWidget'
      >
        <CustomChart
          data={userRole === 'manager' ? dataForPaymentChartManager : dataForPaymentChartResident}
          variant={userRole === 'manager' ? 'paymentManager' : 'paymentResident'}
          title='February overview for all flats:'
          centerText='Total payments'
        />
      </Widget>
    </div>
  );
};

export default PaymentWidget;
