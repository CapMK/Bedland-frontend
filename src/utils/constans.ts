export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
