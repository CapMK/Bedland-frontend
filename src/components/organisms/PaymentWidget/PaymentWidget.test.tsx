import { renderWithProviders } from 'utils/test.utils';
import { fireEvent, screen } from '@testing-library/react';
import routes from 'App/routing/routes';
import { vi } from 'vitest';
import { loggedInManagerUser, loggedInResidentUser } from 'utils/mockUser';
import PaymentWidget from './PaymentWidget';

const mockUseNavigate = vi.fn();
vi.mock('react-router-dom', async () => ({
  ...((await vi.importActual('react-router-dom')) as any),
  useNavigate: () => mockUseNavigate,
}));

describe('Payment Widget ', () => {
  it('should render correctly for manager', () => {
    const widget = renderWithProviders(<PaymentWidget />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    expect(widget).toMatchSnapshot();
  });

  it('should render correctly for resident', () => {
    const widget = renderWithProviders(<PaymentWidget />, {
      preloadedState: {
        user: loggedInResidentUser,
      },
    });
    expect(widget).toMatchSnapshot();
  });

  it('should navigate to paymentPage', async () => {
    renderWithProviders(<PaymentWidget />, {});
    const paymentWidget = await screen.findByRole('button');
    fireEvent.click(paymentWidget);

    expect(mockUseNavigate).toHaveBeenCalled();
    expect(mockUseNavigate).toHaveBeenCalledTimes(1);
    expect(mockUseNavigate).toHaveBeenCalledWith(routes.payments);
  });
});
