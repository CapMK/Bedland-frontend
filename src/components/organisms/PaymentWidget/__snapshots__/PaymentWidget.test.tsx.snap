// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Payment Widget  > should render correctly for manager 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 12.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c6 {
  height: 12.5rem;
  position: relative;
  width: 100%;
}

.c7 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c8 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c9 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c12 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c14 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c13 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c15 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  background-color: #FCFEFF;
  border-radius: 8px;
  box-shadow: 0px 6px 6px -2px rgba(0,0,0,0.15);
  min-height: 22.125rem;
  min-width: 30.875rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: 1px solid #4553DD;
  border-radius: 8px 8px 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.875rem 1.5rem;
}

<body>
    <div>
      <div>
        <div
          class="c0"
          mode="light"
        >
          <div
            class="c1"
            mode="light"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
            >
              Payments
            </h3>
            <button
              class="c2"
            >
              <svg
                data-testid="arrowRight-icon"
                fill="none"
                height="24"
                viewBox="0 0 20 20"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-arrowRight"
                />
                <path
                  d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                  stroke="#4553DD"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.39167"
                />
              </svg>
            </button>
          </div>
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              February overview for all flats:
            </span>
            <div
              class="c5"
              size="12.5"
            >
              <div
                class="c6"
                size="12.5"
              >
                <div
                  class="c7"
                >
                  <canvas
                    height="150"
                    role="img"
                    width="300"
                  />
                </div>
              </div>
              <div
                class="c8"
                id="legendBox"
              >
                <div
                  class="c9"
                >
                  <div
                    class="c9 c10"
                  >
                    Payment overdue
                  </div>
                  <div
                    class="c11"
                  >
                    4%
                  </div>
                </div>
                <div
                  class="c9"
                >
                  <div
                    class="c9 c12"
                  >
                    Awaiting payments
                  </div>
                  <div
                    class="c13"
                  >
                    21%
                  </div>
                </div>
                <div
                  class="c9"
                >
                  <div
                    class="c9 c14"
                  >
                    Paid on time
                  </div>
                  <div
                    class="c15"
                  >
                    75%
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 12.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c6 {
  height: 12.5rem;
  position: relative;
  width: 100%;
}

.c7 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c8 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c9 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c12 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c14 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c13 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c15 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  background-color: #FCFEFF;
  border-radius: 8px;
  box-shadow: 0px 6px 6px -2px rgba(0,0,0,0.15);
  min-height: 22.125rem;
  min-width: 30.875rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: 1px solid #4553DD;
  border-radius: 8px 8px 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.875rem 1.5rem;
}

<div>
    <div>
      <div
        class="c0"
        mode="light"
      >
        <div
          class="c1"
          mode="light"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
          >
            Payments
          </h3>
          <button
            class="c2"
          >
            <svg
              data-testid="arrowRight-icon"
              fill="none"
              height="24"
              viewBox="0 0 20 20"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-arrowRight"
              />
              <path
                d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                stroke="#4553DD"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.39167"
              />
            </svg>
          </button>
        </div>
        <div
          class="c3"
        >
          <span
            class="c4"
          >
            February overview for all flats:
          </span>
          <div
            class="c5"
            size="12.5"
          >
            <div
              class="c6"
              size="12.5"
            >
              <div
                class="c7"
              >
                <canvas
                  height="150"
                  role="img"
                  width="300"
                />
              </div>
            </div>
            <div
              class="c8"
              id="legendBox"
            >
              <div
                class="c9"
              >
                <div
                  class="c9 c10"
                >
                  Payment overdue
                </div>
                <div
                  class="c11"
                >
                  4%
                </div>
              </div>
              <div
                class="c9"
              >
                <div
                  class="c9 c12"
                >
                  Awaiting payments
                </div>
                <div
                  class="c13"
                >
                  21%
                </div>
              </div>
              <div
                class="c9"
              >
                <div
                  class="c9 c14"
                >
                  Paid on time
                </div>
                <div
                  class="c15"
                >
                  75%
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Payment Widget  > should render correctly for resident 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 12.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c6 {
  height: 12.5rem;
  position: relative;
  width: 100%;
}

.c7 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c8 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c9 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c12 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c13 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  background-color: #FCFEFF;
  border-radius: 8px;
  box-shadow: 0px 6px 6px -2px rgba(0,0,0,0.15);
  min-height: 22.125rem;
  min-width: 30.875rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: 1px solid #4553DD;
  border-radius: 8px 8px 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.875rem 1.5rem;
}

<body>
    <div>
      <div>
        <div
          class="c0"
          mode="light"
        >
          <div
            class="c1"
            mode="light"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
            >
              Payments
            </h3>
            <button
              class="c2"
            >
              <svg
                data-testid="arrowRight-icon"
                fill="none"
                height="24"
                viewBox="0 0 20 20"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-arrowRight"
                />
                <path
                  d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                  stroke="#4553DD"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.39167"
                />
              </svg>
            </button>
          </div>
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              February overview for all flats:
            </span>
            <div
              class="c5"
              size="12.5"
            >
              <div
                class="c6"
                size="12.5"
              >
                <div
                  class="c7"
                >
                  <canvas
                    height="150"
                    role="img"
                    width="300"
                  />
                </div>
              </div>
              <div
                class="c8"
                id="legendBox"
              >
                <div
                  class="c9"
                >
                  <div
                    class="c9 c10"
                  >
                    Awaiting payments
                  </div>
                  <div
                    class="c11"
                  >
                    25%
                  </div>
                </div>
                <div
                  class="c9"
                >
                  <div
                    class="c9 c12"
                  >
                    Paid on time
                  </div>
                  <div
                    class="c13"
                  >
                    75%
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 12.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c6 {
  height: 12.5rem;
  position: relative;
  width: 100%;
}

.c7 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c8 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c9 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c12 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c13 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  background-color: #FCFEFF;
  border-radius: 8px;
  box-shadow: 0px 6px 6px -2px rgba(0,0,0,0.15);
  min-height: 22.125rem;
  min-width: 30.875rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: 1px solid #4553DD;
  border-radius: 8px 8px 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.875rem 1.5rem;
}

<div>
    <div>
      <div
        class="c0"
        mode="light"
      >
        <div
          class="c1"
          mode="light"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
          >
            Payments
          </h3>
          <button
            class="c2"
          >
            <svg
              data-testid="arrowRight-icon"
              fill="none"
              height="24"
              viewBox="0 0 20 20"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-arrowRight"
              />
              <path
                d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
                stroke="#4553DD"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.39167"
              />
            </svg>
          </button>
        </div>
        <div
          class="c3"
        >
          <span
            class="c4"
          >
            February overview for all flats:
          </span>
          <div
            class="c5"
            size="12.5"
          >
            <div
              class="c6"
              size="12.5"
            >
              <div
                class="c7"
              >
                <canvas
                  height="150"
                  role="img"
                  width="300"
                />
              </div>
            </div>
            <div
              class="c8"
              id="legendBox"
            >
              <div
                class="c9"
              >
                <div
                  class="c9 c10"
                >
                  Awaiting payments
                </div>
                <div
                  class="c11"
                >
                  25%
                </div>
              </div>
              <div
                class="c9"
              >
                <div
                  class="c9 c12"
                >
                  Paid on time
                </div>
                <div
                  class="c13"
                >
                  75%
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
